src/student/student.entity.ts

```diff
@@ -0,0 +1,151 @@
+import {BaseEntity, Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn} from "typeorm";
+import {Recruiter} from "../recruiter/recruiter.entity";
+import {StudentInterface} from "../types/student";
+
+export enum UserStatus {
+    active = "active",
+    inActive = "inActive",
+    duringTalk = "duringTalk",
+    employed = "employed",
+}
+
+export enum UserRole {
+    admin = "admin",
+    student = "student",
+    recruiter = "recruiter",
+}
+
+export enum TypeWork {
+    stationary = "stationary",
+    readyToMove = "readyToMove",
+    remotely = "remotely",
+    hybrid = "hybrid",
+    noPreference = "noPreference",
+}
+
+export enum ContractType {
+    contractOfEmployment = "contractOfEmployment",
+    b2b = "b2b",
+    contractWorkOrContractOfMandate = "contractWorkOrContractOfMandate",
+    noPreference = "noPreference",
+
+}
+
+@Entity()
+export class Student extends BaseEntity implements StudentInterface {
+    @PrimaryGeneratedColumn('uuid')
+    id: string;
+
+    @Column()
+    token: string;
+
+    @Column()
+    isActive: boolean;
+
+
+    @Column( {
+        unique: true,
+    })
+    @JoinColumn()
+    email: string;
+
+
+    @Column({
+        type: "enum",
+        enum: UserStatus,
+    })
+    status: UserStatus;
+
+    @Column({
+        type: "enum",
+        enum: UserRole,
+    })
+    role: UserRole
+
+    @ManyToOne(type => Recruiter, recruiter => recruiter.id)
+    @JoinColumn({name : "recruiterId"})
+    recruiterId: Recruiter;
+
+    @Column({
+        default: '',
+    })
+    telephone: string;
+
+    @Column({
+        length: 20,
+    })
+    firstName: string;
+
+    @Column({
+        length: 27,
+    })
+    lastName: string;
+
+    @Column({
+        length: 39,
+        unique: true,
+    })
+    githubUsername: string;
```

Comment:
I set length for 39 because thats the longest username Github allows but if we include whole links this may change

Reply:
W specyfikacji jest, żeby sprawdzić API GH lub innym, czy taki suer istnieje. W taki razie ogranicz to do 39 znaków, skoro to max, a trzeba będzie stworzyć dodatkową funkcję do sprawdzenia, czy taki user jest na githubie. 
Możemy to stworzyć po stronie BE albo FE.
W każdym razie w tym polu będzie tylko login z GH